lib/providers/env_var_provider.ex

```diff
@@ -175,13 +188,17 @@ defmodule EnvVar.Provider do
   end
 
   def convert(env_value, :float) do
-    {val, _extra} = Float.parse(env_value)
-    val
+    case Float.parse(env_value) do
+      {value, ""} -> value
+      _other -> raise ArgumentError, "expected float, got: #{inspect(env_value)}"
+    end
   end
 
   def convert(env_value, :integer) do
-    {val, _extra} = Integer.parse(env_value)
-    val
+    case Integer.parse(env_value) do
+      {value, ""} -> value
```

Comment:
We were allowing `123foo` which I think is not good. @relistan wdyt? Were we incidentally using it somewhere afayk?

Reply:
This is a good change. :shipit: